Installation issue: cminpack (invalid c++ compiler)

### Steps to reproduce the issue

```console
$ spack --env zen2-gcc13 find -lvfdIc cminpack
...
-- linux-rhel8-zen2 / %c,fortran=gcc@13.4.0 ---------------------
 -   blniuhd cminpack@1.3.6+blas+ipo+shared build_system=cmake build_type=RelWithDebInfo generator=make patches:=fe6db36
[+]  eufn52p     cmake@3.31.9~doc+ncurses+ownlibs~qtgui build_system=generic build_type=RelWithDebInfo
[+]  gu5vdnc         curl@8.15.0+gssapi~ldap+libidn2~librtmp~libssh+libssh2+nghttp2 build_system=autotools libs:=shared,static tls:=mbedtls,openssl
[+]  r2dc5ck             krb5@1.21.3+shared build_system=autotools
[+]  kdrpmhr                 bison@3.8.2~color build_system=autotools
[+]  vlbkf4y                     m4@1.4.20+sigsegv build_system=autotools
[+]  focymqv                         libsigsegv@2.14 build_system=autotools
[+]  iartorz                 diffutils@3.12 build_system=autotools
[+]  zfcjhbz                     libiconv@1.18 build_system=autotools libs:=shared,static
[+]  zirylds                 findutils@4.10.0 build_system=autotools patches:=440b954
[+]  ful6i3c                 gettext@0.23.1+bzip2+curses+git+libunistring+libxml2+pic+shared+tar+xz build_system=autotools
[+]  btav5ox                     libxml2@2.13.5+http+pic~python+shared build_system=autotools
[+]  ahyoscx                     tar@1.35 build_system=autotools zip=pigz
[+]  vxpii7z                         pigz@2.8 build_system=makefile
[+]  n64duty                         zstd@1.5.7+programs build_system=makefile compression:=lz4,lzma,zlib libs:=shared,static
[+]  3367tmt                             lz4@1.10.0+pic build_system=makefile libs:=shared,static
[+]  ihgv3uq                 libedit@3.1-20240808 build_system=autotools
[+]  xtgeq4a             libidn2@2.3.7 build_system=autotools
[+]  xbdx6lg                 libunistring@1.2 build_system=autotools
[+]  24bzzhf             libssh2@1.11.1+shared build_system=autotools crypto=mbedtls
[+]  pyx73rp                 xz@5.6.3+pic build_system=autotools libs:=shared,static
[+]  crudhjn             mbedtls@2.28.9+pic build_system=makefile build_type=RelWithDebInfo libs:=shared,static
[+]  7wi62dj             nghttp2@1.67.1 build_system=autotools
[+]  unrffxy             openssl@3.6.0~docs+shared build_system=generic certs=system
[+]  edlxjs4             perl@5.42.0+cpanm+opcode+open+shared+threads build_system=generic
[+]  pxa3pju                 berkeley-db@18.1.40+cxx+docs+stl build_system=autotools patches:=26090f4
[+]  jiyqkwo                 bzip2@1.0.8~debug+pic+shared build_system=generic
[+]  m4ifnmh                 gdbm@1.25 build_system=autotools
[+]  f375v3a                     readline@8.3 build_system=autotools patches:=21f0a03
[+]  2pp7u3i             pkgconf@2.5.1 build_system=autotools
[+]  vet42cx         ncurses@6.5-20250705+symlinks+termlib abi=5 build_system=autotools patches:=7a351bc
[+]  p4i3vud         zlib@1.3.1+optimize+pic+shared build_system=makefile
[+]  c4lvqa5     compiler-wrapper@1.0 build_system=generic
[e]  gw3zygo     gcc@13.4.0+binutils~bootstrap+graphite~mold+nvptx~piclibs~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++,fortran,lto'
[+]  gd56l37     gcc-runtime@13.4.0 build_system=generic
[e]  3kzdcra     glibc@2.28 build_system=autotools
[+]  ymb7dyf     gmake@4.4.1~guile build_system=generic
[+]  pn5iw7q     openblas@0.3.30~bignuma~consistent_fpcsr+dynamic_dispatch+fortran~ilp64+locking+pic+shared build_system=makefile symbol_suffix=none threads=openmp
```
(full output [spec.cminpack.txt](https://github.com/user-attachments/files/28369992/spec.cminpack.txt))



### Error message

<details><summary>Error message</summary>

<pre>
...
[ ] blniuhd cminpack@1.3.6 cmake (0s)
[x] blniuhd cminpack@1.3.6 failed: /tmp/swinst26/spackswinst26/spack-stage/spack-stage-cminpack-1.3.6-blniuhdvs4zdd3d3zb6n5ssj5yjxte7e-m2sjk5lj.log (1s)
-- lines 19 to 31 --
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - failed
  -- Check for working CXX compiler: /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/compiler-wrapper-1.0-c4lvqa5ekdyxfibdcfh6hfepzpfknhp5/libexec/spack/gcc/g++
  -- Check for working CXX compiler: /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/compiler-wrapper-1.0-c4lvqa5ekdyxfibdcfh6hfepzpfknhp5/libexec/spack/gcc/g++ - broken
> CMake Error at /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/cmake-3.31.9-eufn52pskigbs2uuke5vjk65qakls5vk/share/cmake-3.31/Modules/CMakeTestCXXCompiler.cmake:73 (message):
    The C++ compiler
 
      "/cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/compiler-wrapper-1.0-c4lvqa5ekdyxfibdcfh6hfepzpfknhp5/libexec/spack/gcc/g++"
  
    is not able to compile a simple test program.
 
-- lines 35 to 57 --
 
      Run Build Command(s): /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/cmake-3.31.9-eufn52pskigbs2uuke5vjk65qakls5vk/bin/cmake -E env VERBOSE=1 /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/gmake-4.4.1-ymb7dyfcziesq5brhndx4pxrqb2g4l25/bin/gmake -f Makefile cmTC_42291/fast
      /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/gmake-4.4.1-ymb7dyfcziesq5brhndx4pxrqb2g4l25/bin/gmake  -f CMakeFiles/cmTC_42291.dir/build.make CMakeFiles/cmTC_42291.dir/build
      gmake[1]: Entering directory '/tmp/swinst26/spackswinst26/spack-stage/spack-stage-cminpack-1.3.6-blniuhdvs4zdd3d3zb6n5ssj5yjxte7e/spack-build-blniuhd/CMakeFiles/CMakeScratch/TryCompile-9ZNxK7'
      Building CXX object CMakeFiles/cmTC_42291.dir/testCXXCompiler.cxx.o
      /cvmfs/hpcsw.umd.edu/spack-software/2025.11.0/linux-zen2/compiler-wrapper-1.0-c4lvqa5ekdyxfibdcfh6hfepzpfknhp5/libexec/spack/gcc/g++    -o CMakeFiles/cmTC_42291.dir/testCXXCompiler.cxx.o -c /tmp/swinst26/spackswinst26/spack-stage/spack-stage-cminpack-1.3.6-blniuhdvs4zdd3d3zb6n5ssj5yjxte7e/spack-build-blniuhd/CMakeFiles/CMakeScratch/TryCompile-9ZNxK7/testCXXCompiler.cxx
>     [spack cc]: Error: SPACK_CXX_* variables not set: this usually means a missing `depends_on("cxx", type="build")` in package.py
>     gmake[1]: *** [CMakeFiles/cmTC_42291.dir/build.make:81: CMakeFiles/cmTC_42291.dir/testCXXCompiler.cxx.o] Error 1
      gmake[1]: Leaving directory '/tmp/swinst26/spackswinst26/spack-stage/spack-stage-cminpack-1.3.6-blniuhdvs4zdd3d3zb6n5ssj5yjxte7e/spack-build-blniuhd/CMakeFiles/CMakeScratch/TryCompile-9ZNxK7'
>     gmake: *** [Makefile:134: cmTC_42291/fast] Error 2
 
...
</pre></details>


### Information on your system

```console
 spack debug report
* **Spack:** 1.2.0.dev0
* **Builtin repo:** https://github.com/spack/spack-packages/commit/119680aeee8ea802c6111b7167583bddef97e82f
* **Python:** 3.6.8
* **Platform:** linux-rhel8-zen2
```

### Additional information

- [spack-build-out.txt](https://github.com/user-attachments/files/28369916/spack-build-out.txt)
- [spack-build-env.txt](https://github.com/user-attachments/files/28369923/spack-build-env.txt)

No maintainers found.

The package.py for cminpack declares a dependency on c but not on cxx, which appears to be needed (at least by the cmake infrastructure).

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/cminpack/package.py
+++ b/repos/spack_repo/builtin/packages/cminpack/package.py
@@ -23,9 +23,9 @@ class Cminpack(CMakePackage):
     variant("shared", default=False, description="Build shared libraries")
     variant("blas", default=True, description="Compile with BLAS")
 
-    depends_on("c", type="build")  # generated
-    depends_on("fortran", type="build")  # generated
-    depends_on("cxx", type="build")  # Fixes issue #5012
+    depends_on("c", type="build")
+    depends_on("cxx", type="build")
+    depends_on("fortran", type="build")
 
     depends_on("blas", when="+blas")
 

--- a/repos/spack_repo/builtin/packages/cminpack/package.py
+++ b/repos/spack_repo/builtin/packages/cminpack/package.py
@@ -25,7 +25,7 @@ class Cminpack(CMakePackage):
 
     depends_on("c", type="build")  # generated
     depends_on("fortran", type="build")  # generated
-    depends_on("cxx", type="build") #Fixes issue #5012
+    depends_on("cxx", type="build")  # Fixes issue #5012
 
     depends_on("blas", when="+blas")
 

--- a/repos/spack_repo/builtin/packages/cminpack/package.py
+++ b/repos/spack_repo/builtin/packages/cminpack/package.py
@@ -25,6 +25,7 @@ class Cminpack(CMakePackage):
 
     depends_on("c", type="build")  # generated
     depends_on("fortran", type="build")  # generated
+    depends_on("cxx", type="build") #Fixes issue #5012
 
     depends_on("blas", when="+blas")
 

